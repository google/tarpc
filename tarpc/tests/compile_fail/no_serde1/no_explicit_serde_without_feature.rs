#[tarpc::service(derive_serde = true)]
trait Foo {
    async fn foo();
}

fn main() {
    let x = FooRequest::Foo {};
    x.serialize();
}
