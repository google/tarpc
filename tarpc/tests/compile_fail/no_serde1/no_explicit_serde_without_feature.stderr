error: To enable serde, first enable the `serde1` feature of tarpc
 --> tests/compile_fail/no_serde1/no_explicit_serde_without_feature.rs:1:18
  |
1 | #[tarpc::service(derive_serde = true)]
  |                  ^^^^^^^^^^^^

error[E0433]: failed to resolve: use of undeclared type `FooRequest`
 --> tests/compile_fail/no_serde1/no_explicit_serde_without_feature.rs:7:13
  |
7 |     let x = FooRequest::Foo {};
  |             ^^^^^^^^^^ use of undeclared type `FooRequest`
